src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -502,14 +510,108 @@ ReportAie::getPropertyTree20202(const xrt_core::device * _pDevice,
   _pt.add_child("aie_metadata", populate_aie(_pDevice, "Aie_Metadata"));
 }
 
-void 
+void
+AieTopThreadFunction(const boost::property_tree::ptree& _pt)
+{
+  std::ostream &_output = std::cout;
+
+  while(!is_quit)
+  {
+    try
+    {
+#if defined _WIN32
+      system("cls");
+#elif defined (__LINUX__) || defined(__gnu_linux__) || defined(__linux__)
+      system("clear");
+#endif
+      // validate and print aie metadata by checking schema_version node
+      if(!_pt.get_child_optional("aie_metadata.schema_version"))
+        return;
+
+      _output << "Aie\n";
+      _output << boost::format("  %-10s\n") % _pt.get<std::string>("aie_metadata.description");
+
+      for (auto& gr: _pt.get_child("aie_metadata.graphs")) {
+        const boost::property_tree::ptree& graph = gr.second;
+
+        _output << boost::format("  Graph [  #]  %-30s%-30s\n") % "Name" % "Status";
+	_output << boost::format("        [ %2d]  %-30s%-30s\n") % graph.get<std::string>("id")
+		% graph.get<std::string>("name") % graph.get<std::string>("status");
+        _output << std::endl;
+        _output << boost::format("  %s\n") % "Cores";
+        _output << boost::format("  Core  [  #]  %-30s%-20s%-20s%-20s%-20s\n") % "[C:R]"
+	           % "Status" % "Program Counter" % "Link Register" % "Stack Pointer";
+
+	int count = 0;
+        for (auto& tile : graph.get_child("tile")) {
+          _output << boost::format("        [ %2d]  %-30s%-20s%-20s%-20s%-20s\n") % count++ 
+		% (tile.second.get<std::string>("column") + ":" + tile.second.get<std::string>("row"))
+		% tile.second.get<std::string>("core.status")
+		% tile.second.get<std::string>("core.program_counter")
+		% tile.second.get<std::string>("core.link_register")
+		% tile.second.get<std::string>("core.stack_pointer");
+        }
+      }
+
+      // Sleep and check for interrupt.
+      std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    catch(std::exception const& e)
+    {
+      _output <<  e.what() << std::endl;
+    }
+  }
+}
+
+void
+sig_handler(const boost::system::error_code& error, int signal_number)
+{
+  is_quit = true;
+  std::cout << "Top command exiting... "  << std::endl;
+}
+
+void
+AieTopFunc(const boost::property_tree::ptree& _pt)
+{
+  boost::asio::io_service io_service;
+
+  is_quit = false;
+  // Create a threadmain function for printing the status
+  std::thread t_thread(AieTopThreadFunction, _pt);

```

Comment:
This is a major architectural use-case change.  Reports are currently designed to take a one (1) sample reading and report the results.  This "threading" concept (e.g., equivalent to xbutil top), is is currently scheduled for 2022.1.

Reply:
This functionality is removed. We will take care this portion in an another story for 2022.1 https://jira.xilinx.com/browse/VITIS-3365